### ISSUE TITLE
larger lineheight gives lower accuracy, anyone know why?

### ISSUE THREAD

**Issue by ocrwork** [Calamari-OCR/calamari#140]
I have 1100 lines of training data, with each line image 144 high by 3864 wide.
If I train with batch 20, lineheight 92, I get validation accuracy of 97% at iteration 3000.
If I lower the lineheight to 92 or 88, accuracy drops off a few percent.
If I increase the lineheight to 144, validation accuracy never gets above 35%, even after 90000 iterations.
Can anyone tell me if I should be adjusting some network parameters, by that I mean to not use the default network architecture? Or is there somthing else to adjust to get better accuracy at lineheight 144?
Thanks very much.

**Comment by ChWick** [comment]
* Is there a reason, why you require a line height of 144?
* Is must be expected that larger line heights result in a drop of accuracy because more information (higher number of pixels) is fed into the network which then can easier overfit. Furthermore, for a prediction the network has a "smaller area of interest", i.e., a letter might be so large, that the CNN/LSTM has no access to the full widespread information required for a single prediction.
* You could try to add another CNN/Pooling layer, which reduces the dimensions for the LSTM layer: `cnn=40:3,pool=2,cnn=60:3,pool=2,cnn=80:3,pool=2,lstm=200,dropout=0.5`, but maybe a fewer number of channels is better: `cnn=20:3,pool=2,cnn=40:3,pool=2,cnn=60:3,pool=2,lstm=200,dropout=0.5`

**Comment by ocrwork** [comment]
I am working with an asian character set, which has various marks attached above and below, so it gets pretty complex. I thought more detail might improve the accuracy. On the other hand, training on the same character set taken from a pdf made from a typeset font, it gets 99.97% validation set accuracy using lineheight 96. I will try out the CNN/Pooling layer ideas, thank you!

**Comment by deepseek** [comment]
@ocrwork post back your results here.

**Comment by ocrwork** [comment]
results:

1128 lines in dataset
24 lines in validation set
the extra cnn/pool layers made it work well with line_height 144. Thanks!
But the extra layers made the line_height 96 fail to converge at all.
Also, my idea that the accuracy would increase by using larger line_height appears to
be incorrect. I am guessing I need more lines in the training data. I have seen that 5000
lines gives much better accuracy than lower numbers.
So, I learn that the optimal number of cnn/pool layers is correlated with 
line_height. Makes sense. It seems like the optimal line_height for each network is:

cnn/pool layers   line_height
2                         96
3                          144

training results:
--network cnn=40:3,pool=2,cnn=60:3,pool=2,cnn=80:3,pool=2,lstm=200,dropout=0.5 \
batch_size 20 \
line_height 144
iter 400 92.45% accuracy
iter 800 95.32% accuracy
iter 1200 95.54% accuracy
iter 1600 96.14% accuracy
iter 2400 96.43% accuracy
iter 4000 96.5% accuracy
iter 5600 96.94% accuracy
iter 15200 96.96% accuracy

--network cnn=20:3,pool=2,cnn=40:3,pool=2,cnn=60:3,pool=2,lstm=200,dropout=0.5
batch_size 20 \
line_height 144
iter 400 91.44%
iter 800 93.58%
iter 1200 94.98%
iter 2800 95.43%
nothing better up to iter 50000.
